src/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/CustomValidationAttribute.cs

```diff
@@ -184,12 +184,7 @@ protected override ValidationResult IsValid(object value, ValidationContext vali
             }
             catch (TargetInvocationException tie)
             {
-                if (tie.InnerException != null)
-                {
-                    throw tie.InnerException;
-                }
-
-                throw;
+                throw tie.InnerException;
```

Comment:
This is the `try` block:

```
try
{
    // 1-parameter form is ValidationResult Method(object value)
    // 2-parameter form is ValidationResult Method(object value, ValidationContext context),
    var methodParams = _isSingleArgumentMethod
        ? new object[] { convertedValue }
        : new[] { convertedValue, validationContext };

    var result = (ValidationResult)methodInfo.Invoke(null, methodParams);

    // We capture the message they provide us only in the event of failure,
    // otherwise we use the normal message supplied via the ctor
    _lastMessage = null;

    if (result != null)
    {
        _lastMessage = result.ErrorMessage;
    }

    return result;
}
catch (TargetInvocationException tie)
{
    throw tie.InnerException;
}
```
The only place where the TIE can be thrown is from reflection invoke, and there will always be a non-null inner exception in this case.